dao/chat.go

```diff
@@ -164,7 +164,9 @@ func (d chatDao) RetractChat(id uint) error {
 
 // DeleteChat removes a chat with the given id from the database.
 func (d chatDao) DeleteChat(id uint) error {
-	return DB.Model(&model.Chat{}).Delete(&model.Chat{}, id).Error
+	return errors.Join(
+		DB.Model(&model.Chat{}).Delete(&model.Chat{}, id).Error,
+		DB.Delete(&model.Chat{}, "reply_to = ?", id).Error)
```

Comment:
This seems a little cleaner

```suggestion
	return DB.Where("id = ? OR reply_to = ?").Delete(&model.Chat{}).Error
```